src/routes_api.py

```diff
@@ -51,4 +57,204 @@ def trigger_scheduled_run(cid, aid, scheduled_run_id):
                 db.update_scheduled_run_status(sched_run["_id"], ScheduledRunStatus.RAN)
                 db.update_scheduled_run_bw_run_id(sched_run["_id"], bw_run_id)
             return util.success("")
+        
+        # Want to avoid stuff like this, with overlaps in function definitions
+        # Best way is to consider an AdminOperations class and have AdminRoutes and APIRoutes
+        # use the functionality defined in there, instead of whatever I did with AdminRoutes currently
+        @blueprint.route("/api/<cid>/<aid>/add_extension", methods=["POST"])
+        @auth.require_course_auth
+        @auth.require_admin_status
+        def add_extension(cid, aid):
+            form = request.json
+            
+            assignment = db.get_assignment(cid, aid)
+            if not assignment:
+                return util.error("Invalid course or assignment.\nPlease try again.")
+            
+        
+            missing = util.check_missing_fields(form, "netids", "max_runs", "start", "end")
```

Comment:
If the user neglects to add a config when adding extensions with a grading run, the app will throw an exception and have to be restarted (which happens automatically, but still). We should probably check to make sure the config is present before calling `add_or_edit_scheduled_run`. Obviously, please correct me if I'm wrong on this.

Reply:
`add_or_edit_scheduled_run` should check for that already, no? https://github.com/illinois-cs241/broadway-on-demand/blob/e8a90865c2cab933c97f401c3a1628f0296605eb/src/routes_api.py#L187 What input did you put in to throw this error?